### PR TITLE
[JENKINS-41306] Migrate to 2.x parent POM. Fix javadoc issues.

### PR DESCRIPTION
**Pull request by varyvol**
[JENKINS-41306](https://issues.jenkins-ci.org/browse/JENKINS-41306)

- Migrate to 2.x parent POM.
- Fix some Javadoc issues.
- Fix some Findbugs issues.

@reviewbybees @tfennelly

### CODE REVIEW

**Comment by kwhetstone** [comment]
🐝

**Review by andresrc — APPROVED** [review]
:bee:

File: pom.xml [hunk 1]
```diff
@@ -70,6 +70,11 @@ THE SOFTWARE.
...
   </developers>
 
+  <properties>
+    <jenkins.version>1.609</jenkins.version>
+    <java.level>6</java.level>
+  </properties>
+
   <build>
     <plugins>
       <plugin>
```
Comment (andresrc) [inline pom.xml:80]: * The version for the `maven-release-plugin` can be obtained from the parent pom
* Change `http` to `https` in the repositories.

**Review by andresrc — CHANGES_REQUESTED** [review]

**Comment by varyvol** [comment]
@andresrc changes done. For the maven-release-plugin, I removed the version so the one from the parent POM is gathered. Is that what you meant?

**Comment by varyvol** [comment]
@jglick I think you have also put some work on this plugin, so your review would be much appreciated.

**Review by andresrc — APPROVED** [review]

**Comment by andresrc** [comment]
@tfennelly @jglick ok to merge and release this?

**Comment by ghost** [comment]
This pull request originates from a [CloudBees](https://www.cloudbees.com/) employee. At CloudBees, we require that all pull requests be reviewed by other CloudBees employees before we seek to have the change accepted. If you want to learn more about our process please see [this explanation](https://github.com/reviewbybees/about#about-reviewbybees).

File: icon-set/src/main/java/org/jenkins/ui/icon/Icon.java [hunk 2]
```diff
@@ -25,6 +25,7 @@
 
 import org.apache.commons.jelly.JellyContext;
 
+import java.io.Serializable;
```
Comment (tfennelly) [inline icon-set/src/main/java/org/jenkins/ui/icon/Icon.java:28]: I don't see any other change relating to this?

**Review by tfennelly — APPROVED** [review]
LGTM 👍

File: icon-set/src/main/java/org/jenkins/ui/icon/Icon.java (hunk 2 above)
Comment (varyvol) [reply icon-set/src/main/java/org/jenkins/ui/icon/Icon.java:28]: The StringComparator implements it.

**Pull request merged**